src/main/java/org/magicdgs/popgenlib/utils/FrequencyUtils.java

```diff
@@ -0,0 +1,148 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.magicdgs.popgenlib.utils;
+
+import org.apache.commons.math3.util.Pair;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Utility methods for working with frequencies. If frequencies are not valid, this methods throws
+ * an {@link IllegalFrequencyException}.
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public class FrequencyUtils {
+
+    /** Maximum frequency. */
+    public static final Double FREQUENCY_ONE = 1d;
+
+    /** Minimum frequency. */
+    public static final Double FREQUENCY_ZERO = 0d;
+
+    // cannot be instantiated
+    private FrequencyUtils() {}
+
+    /**
+     * Validates that the frequencies are correct:
+     * <ul>
+     * <li>Frequency list should not be empty.</li>
+     * <li>Frequencies should be in the range [{@link #FREQUENCY_ZERO},
+     * {@link #FREQUENCY_ONE}].</li>
+     * <li>Frequencies should sum exactly {@link #FREQUENCY_ONE}.</li>
+     * </ul>
+     *
+     * @param frequencies frequencies to validate.
+     *
+     * @throws IllegalFrequencyException if frequencies are not valid.
+     */
+    public static void validateFrequencies(final List<Double> frequencies) {
+        try {
+            final double sum = Verify.nonEmpty(frequencies, () -> "frequencies")
+                    .stream().mapToDouble(FrequencyUtils::validateRange)
+                    .sum();
+            Verify.validate(FREQUENCY_ONE.equals(sum),
+                    () -> String.format("Frequencies should sum 1 but found %s: %s",
+                            sum, frequencies));
+        } catch (IllegalArgumentException e) {
+            throw new IllegalFrequencyException(e);
+        }
+    }
+
+    // validate the range and return the frequency cast as a double
+    private static double validateRange(final Double freq) {
+        try {
+            Verify.validate(freq >= FREQUENCY_ZERO && freq <= FREQUENCY_ONE,
+                    () -> String.format("Frequencies out of range [%s, %s]: %s",
+                            FREQUENCY_ZERO, FREQUENCY_ONE, freq));
+            return freq;
+        } catch (IllegalArgumentException e) {
```

Comment:
Here the catching could be removed, because it should be done outside this method.